### PR TITLE
[GH-755] ChannelStateOnChain adjustments

### LINKED ISSUE #755: Adjust ChannelStateOnChain structure
Adjust fields:
This does **not include**: handling of delegates, handling of new txs (force progress, snapshot)
This **includes** relevant changes to transactions so everything still works correctly.

### PR DESCRIPTION
**Pull request by cytadela8**
- adjusted ChannelStateOnChain structure
- adjusted serialization
- adjusted mutal close check
- added identifier value list encoders
- disabled old compatibility test

This closes #755 
*Before asking questions about scope of this PR please see the issue description.*

### CODE REVIEW

File: apps/aecore/lib/aecore/chain/identifier.ex [hunk 1]
```diff
@@ -78,6 +87,22 @@ defmodule Aecore.Chain.Identifier do
     end
   end
 
+  @spec decode_from_binary_list_to_value_list(list(binary()), type()) ::
+          list(value()) | {:error, String.t()}
```
Comment (DanielaIvanova) [inline apps/aecore/lib/aecore/chain/identifier.ex]: Could you fix the specification of this function?

**Review by DanielaIvanova — DISMISSED** [review]

**Review by DanielaIvanova — APPROVED** [review]

**Review by gorbak25 — APPROVED** [review]
Looks good.

File: apps/aecore/lib/aecore/channel/channel_state_on_chain.ex [hunk 2]
```diff
@@ -45,24 +47,28 @@ defmodule Aecore.Channel.ChannelStateOnChain do
...
   - initiator_pubkey
   - responder_pubkey
+  - delegates - list of delegates alowed to perform certain operations
   - total_amount - the total amount of tokens in the channel
   - initiator_amount - amount deposited by initiator in create_tx or from poi
   - responder_amount - amount deposited by responder in create_tx or from poi
   - lock_period - time before slashing is settled
   - closing_at - when != 0: block height when channel will be irrevocably closed
   - sequence - sequence of highest known fully signed offchain chainstate
+  - solo_sequence - when !=0: sequence of first force progresses
```
Comment (grzegorz225) [inline apps/aecore/lib/aecore/channel/channel_state_on_chain.ex]: progresses -> progress

File: apps/aecore/lib/aecore/channel/channel_state_on_chain.ex [hunk 3]
```diff
@@ -45,24 +47,28 @@ defmodule Aecore.Channel.ChannelStateOnChain do
   # Parameters
   - initiator_pubkey
   - responder_pubkey
+  - delegates - list of delegates alowed to perform certain operations
```
Comment (grzegorz225) [inline apps/aecore/lib/aecore/channel/channel_state_on_chain.ex]: alowed -> allowed

**Review by thepiwo — APPROVED** [review]
looks great

**Pull request merged**